poh/src/poh_recorder.rs

```diff
@@ -569,6 +569,15 @@ impl PohRecorder {
             self.leader_first_tick_height_including_grace_ticks
         {
             if self.reached_leader_tick(my_pubkey, leader_first_tick_height_including_grace_ticks) {
+                if self.blockstore.has_existing_shreds_for_slot(next_poh_slot) {
```

Comment:
This is the only functional change. 
This fixes a long standing bug where in certain rare scenarios we will attempt to produce a block that we have already produced.

This does not transmit a duplicate block, as a failsafe in `broadcast_stage` will prevent us from sending the shreds out, however it results in wasted effort in `banking_stage` as the block is reproduced.